### ISSUE TITLE
MNELAB menus don't work on macOS

### ISSUE THREAD

**Issue by cbrnr** [conda-forge/mnelab-feedstock#10]
I just installed MNELAB with `conda create -y -n mnelab -c conda-forge mnelab`. When I then run MNELAB, the menus don't work (they are shown, but clicking on them has no effect). Otherwise, the app seems to work because I can e.g. open a file with the toolbar icon.

Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/clemens/anaconda/envs/mnelab:
#
# Name                    Version                   Build  Channel
apptools                  4.5.0                      py_0    conda-forge
blosc                     1.17.1               h4a8c4bd_0    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2019.11.28           hecc5488_0    conda-forge
certifi                   2019.11.28               py38_0    conda-forge
configobj                 5.0.6                      py_0    conda-forge
curl                      7.65.3               h22ea746_0    conda-forge
cvxopt                    1.2.4           py38hd42b0c3_201    conda-forge
cycler                    0.10.0                     py_2    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
dipy                      1.1.0            py38h3b54f70_0    conda-forge
dsdp                      5.8               h7674d01_1203    conda-forge
envisage                  4.8.0                      py_0    conda-forge
expat                     2.2.9                h4a8c4bd_1    conda-forge
fftw                      3.3.8           nompi_h44ae4c8_1110    conda-forge
freetype                  2.10.0               h24853df_1    conda-forge
future                    0.18.2                   py38_0    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
glib                      2.58.3          py38h577aef8_1002    conda-forge
glpk                      4.65              h65ac59c_1002    conda-forge
gmp                       6.2.0                h4a8c4bd_0    conda-forge
gsl                       2.5                  ha2d443c_1    conda-forge
h5py                      2.10.0          nompi_py38h106b333_101    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.5          nompi_h3e39495_1104    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
imageio                   2.6.1                    py38_0    conda-forge
joblib                    0.14.1                     py_0    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
jsoncpp                   1.8.4             ha1b3eb9_1002    conda-forge
kiwisolver                1.1.0            py38ha1b3eb9_0    conda-forge
krb5                      1.16.4               h1752a42_0    conda-forge
libblas                   3.8.0               14_openblas    conda-forge
libcblas                  3.8.0               14_openblas    conda-forge
libclang                  8.0.1                h770b8ee_1    conda-forge
libcurl                   7.65.3               h16faf7d_0    conda-forge
libcxx                    9.0.1                         1    conda-forge
libedit                   3.1.20170329      hcfe32e1_1001    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblapack                 3.8.0               14_openblas    conda-forge
libllvm8                  8.0.1                h770b8ee_0    conda-forge
libnetcdf                 4.7.3           nompi_hda4e5f1_101    conda-forge
libopenblas               0.3.7                h3d69b6c_7    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libssh2                   1.8.2                hcdc9a53_2    conda-forge
libtiff                   4.1.0                ha78913b_3    conda-forge
libxml2                   2.9.10               h53d96d6_0    conda-forge
llvm-openmp               9.0.1                h28b9765_2    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
matplotlib                3.1.2                    py38_1    conda-forge
matplotlib-base           3.1.2            py38h11da6c2_1    conda-forge
mayavi                    4.7.1            py38ha91d4f2_2    conda-forge
metis                     5.1.0             h4a8c4bd_1005    conda-forge
mne                       0.19.2                     py_1    conda-forge
mnelab                    0.5.2                      py_0    conda-forge
mock                      3.0.5                    py38_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
nibabel                   3.0.1                      py_0    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numexpr                   2.7.1            py38h4f17bb1_0    conda-forge
numpy                     1.16.5           py38hde6bac1_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openssl                   1.1.1d               h0b31af3_0    conda-forge
pcre                      8.43                 h4a8c4bd_0    conda-forge
pillow                    7.0.0            py38h918e99a_0    conda-forge
pip                       20.0.2                   py38_0    conda-forge
pybv                      0.2.0                      py_0    conda-forge
pydicom                   1.4.1                      py_0    conda-forge
pyedflib                  0.1.15           py38h3b54f70_0    conda-forge
pyface                    6.1.2                      py_0    conda-forge
pygments                  2.5.2                      py_0    conda-forge
pyparsing                 2.4.6                      py_0    conda-forge
pyqt                      5.12.3           py38he22c54c_1    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pysurfer                  0.9.0              pyh24bf2e0_0    conda-forge
pytables                  3.6.1            py38h6f8395a_1    conda-forge
python                    3.8.1                h5c2c468_1    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-picard             0.4                pyh5ca1d4c_0    conda-forge
pyxdf                     1.16.2                     py_0    conda-forge
qt                        5.12.5               h1b46049_0    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
scikit-learn              0.22.1           py38h3dc85bc_1    conda-forge
scipy                     1.4.1            py38h82752d6_0    conda-forge
setuptools                45.1.0                   py38_0    conda-forge
six                       1.14.0                   py38_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
suitesparse               5.6.0                h0e59142_0    conda-forge
tbb                       2019.9               ha1b3eb9_1    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
tornado                   6.0.3            py38h0b31af3_0    conda-forge
traits                    5.2.0            py38h0b31af3_0    conda-forge
traitsui                  6.1.3                      py_0    conda-forge
vtk                       8.2.0           py38hb651df3_207    conda-forge
wheel                     0.34.1                   py38_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                he7fca8b_1    conda-forge
```
</details>

Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : mnelab
    active env location : /Users/clemens/anaconda/envs/mnelab
            shell level : 1
       user config file : /Users/clemens/.condarc
 populated config files : /Users/clemens/.condarc
          conda version : 4.8.1
    conda-build version : not installed
         python version : 3.7.6.final.0
       virtual packages : __osx=10.15.2
       base environment : /Users/clemens/anaconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/clemens/anaconda/pkgs
                          /Users/clemens/.conda/pkgs
       envs directories : /Users/clemens/anaconda/envs
                          /Users/clemens/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.1 requests/2.22.0 CPython/3.7.6 Darwin/19.2.0 OSX/10.15.2
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</details>

**Comment by hoechenberger** [comment]
Thanks for the report, I will look into this

**Comment by hoechenberger** [comment]
@cbrnr If if Cmd-Tab to another app and then Cmd-Tab back to MNELAB again, the menus work. Very odd…

**Comment by hoechenberger** [comment]
So this seems to be a quite common problem.

Just leaving this here for reference:
https://github.com/napari/napari/issues/380#issuecomment-567281044

**Comment by cbrnr** [comment]
The strange thing is that it *does* work on the same machine without conda (i.e. Homebrew Python, mostly identical package versions). I've had issues with conda and Qt in the past, so this might be yet another problem. Another issue is that the entry at the top left isn't MNELAB, but python3.8 - pip installing `pyobjc-framework-cocoa` doesn't help either (it seems like there aren't any conda-forge packages). I'm thinking if it is a good idea to have conda-forge instructions in the MNELAB installation instructions when we know it doesn't work on macOS...

**Comment by hoechenberger** [comment]
@cbrnr I'll leave this up to you, of course :) I will keep investigating the issue and can offer to keep you posted!

**Comment by hoechenberger** [comment]
@cbrnr State of my current research results is:
This seems to be an issue that _many_ people report, even when just using Qt (i.e., it's not specific to PyQt) on macOS. Only way to reliably work around this is to create a `.app` bundle. I'm surprised it's working for you with homebrew, though!

**Comment by cbrnr** [comment]
I never had problems with Qt on my Mac, including PyQt/PySide2. However, every time I try to use conda (especially in combination with Qt, but not exclusively) something doesn't work - that's why I stick with Homebrew Python, it just works much better in my experience. I can't find the issues I had off the top of my head, I'm not even sure I reported all of them (because I immediately switched back to Homebrew Python). Some of them had to do with Python not being a framework build in Anaconda, which Qt didn't like. Not sure if this is related or if it has been resolved so far.

Can you point me to similar Qt issues that do *not* involve conda? I could try and see if these things work on my setup.

**Comment by hoechenberger** [comment]
> Can you point me to similar Qt issues that do _not_ involve conda? I could try and see if these things work on my setup.

Yes, will do that later tonight :)

**Comment by hoechenberger** [comment]
@cbrnr If I run `mnelab` through `pythonw` (which is installed through the `python.app` package), all works well. Currently trying to fix the `mnelab` entry point on macOS in #11

As for starting MNELAB via `python -m mnelab`, we could simply add a small workaround in MNELAB itself if that would be okay with you; I've done this before for PsychoPy. Idea is to figure out if the current Python interpreter is running on macOS, and if yes, spawn a new process via `pythonw`. I could file a PR upstream. Let me know what you think!

**Comment by hoechenberger** [comment]
The entry point was fixed via #11, #12, and #13.

Invocation via `python -m mnelab` will start working once https://github.com/cbrnr/mnelab/pull/119 gets merged upstream.

**Comment by cbrnr** [comment]
https://github.com/cbrnr/mnelab/pull/119 was merged so closing this.
